### PR TITLE
Fix #36: Remove yiisoft/view dependency

### LINKED ISSUE #36: Remove yiisoft/view since there will be only API

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #36

### CODE REVIEW

File: config/routes.php [hunk 1]
```diff
@@ -6,5 +6,45 @@
...
+use Psr\Http\Message\ResponseFactoryInterface;
+use Yiisoft\DataResponse\Middleware\FormatDataResponseAsJson;
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+use Yiisoft\Validator\ValidatorInterface;
+use Yiisoft\Yii\Debug\Api\Middleware\ResponseDataWrapper;
+use Yiisoft\Yii\Gii\Controller\DefaultController;
+use Yiisoft\Yii\Middleware\IpFilter;
+
+if (!(bool)($params['yiisoft/yii-enabled']['enabled'] ?? false)) {
```
Comment (arogachev) [inline config/routes.php:18]: Seems like there is a typo in `yii-enabled`?

Comment (rustamwin) [reply config/routes.php]: Fixed

**Review by xepozz — APPROVED** [review]

File: src/Controller/DefaultController.php [hunk 2]
```diff
@@ -4,180 +4,108 @@
...
-                    $params['hasError'] = !$generator->save($files, (array)$answers, $results);
-                    $params['results'] = $results;
-                } else {
-                    $params['files'] = $files;
-                    $params['answers'] = $answers;
-                }
-            }
+    public function generate(GeneratorRequest $request): ResponseInterface
+    {
+        $generator = $request->getGenerator();
```
Comment (arogachev) [inline src/Controller/DefaultController.php:39]:
```suggestion
        /** @var AbstractGenerator $generator */
        $generator = $request->getGenerator();
```

Comment (arogachev) [reply src/Controller/DefaultController.php:39]: Fixes Psalm error.

File: src/Generator/AbstractGenerator.php [hunk 3]
```diff
@@ -337,15 +328,34 @@ public function getTemplatePath(): string
...
      */
     public function render(string $template, array $params = []): string
     {
-        $params['generator'] = $this;
-
-        return $this->view->withContext($this)->render($template, $params);
+        $file = sprintf('%s/%s.php', $this->aliases->get($this->getTemplatePath()), $template);
+
+        $renderer = function (): void {
+            extract(array: func_get_arg(1), flags: EXTR_OVERWRITE);
```
Comment (arogachev) [inline src/Generator/AbstractGenerator.php:339]:
```suggestion
            $array = func_get-arg(1);
            extract($array, flags: EXTR_OVERWRITE);
```

Comment (arogachev) [reply src/Generator/AbstractGenerator.php:339]: Fixes Psalm error.

File: src/Request/GeneratorRequest.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+final class GeneratorRequest extends RequestModel
+{
+    public function __construct(private GiiInterface $gii)
+    {
+    }
+
+    public function getGenerator(): GeneratorInterface
+    {
+        $generator = $this->gii->getGenerator($this->getAttributeValue('router.generator'));
```
Comment (arogachev) [inline src/Request/GeneratorRequest.php:19]:
```suggestion
        /** @var AbstractGenerator $generator */
        $generator = $this->gii->getGenerator($this->getAttributeValue('router.generator'));
```

Comment (arogachev) [reply src/Request/GeneratorRequest.php:19]: Fixes Psalm error.

File: src/Request/GeneratorRequest.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Gii\Request;
+
+use Yiisoft\RequestModel\RequestModel;
+use Yiisoft\Yii\Gii\GeneratorInterface;
+use Yiisoft\Yii\Gii\GiiInterface;
```
Comment (arogachev) [inline src/Request/GeneratorRequest.php:9]:
```suggestion
use Yiisoft\RequestModel\RequestModel;
use Yiisoft\Yii\Gii\Generator\AbstractGenerator;
use Yiisoft\Yii\Gii\GeneratorInterface;
use Yiisoft\Yii\Gii\GiiInterface;
```

File: src/Request/GeneratorRequest.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Gii\Request;
+
+use Yiisoft\RequestModel\RequestModel;
```
Comment (arogachev) [inline src/Request/GeneratorRequest.php:7]:
```suggestion
use Yiisoft\RequestModel\RequestModel;
use Yiisoft\Yii\Gii\Generator\AbstractGenerator;
```

Comment (arogachev) [reply src/Request/GeneratorRequest.php:7]: An addition to https://github.com/yiisoft/yii-gii/pull/56/files#r993318415.

File: src/CodeFile.php [hunk 7]
```diff
@@ -34,6 +33,12 @@ final class CodeFile
      * The new code file and the existing one are identical.
      */
     public const OP_SKIP = 2;
+    public const OPERATIONS = [
```
Comment (arogachev) [inline src/CodeFile.php:36]:
```suggestion
    public const OPERATIONS_MAP = [
```

File: config/routes.php [hunk 8]
```diff
@@ -6,5 +6,45 @@
  * @var array $params
  */
 
+use Psr\Http\Message\ResponseFactoryInterface;
+use Yiisoft\DataResponse\Middleware\FormatDataResponseAsJson;
+use Yiisoft\Router\Group;
```
Comment (arogachev) [inline config/routes.php:11]: Undefined class. Need to add `yiisoft/router` to dependencies.

File: config/routes.php [hunk 9]
```diff
@@ -6,5 +6,45 @@
...
+use Yiisoft\Router\Route;
+use Yiisoft\Validator\ValidatorInterface;
+use Yiisoft\Yii\Debug\Api\Middleware\ResponseDataWrapper;
+use Yiisoft\Yii\Gii\Controller\DefaultController;
+use Yiisoft\Yii\Middleware\IpFilter;
```
Comment (arogachev) [inline config/routes.php:16]: Undefined class.

File: config/routes.php [hunk 10]
```diff
@@ -6,5 +6,45 @@
...
+             }
+         )
+         ->middleware(FormatDataResponseAsJson::class)
+         ->middleware(ResponseDataWrapper::class)
+         ->namePrefix('gii/api/')
+         ->routes(
+             Route::get('/{generator}')
+                  ->action([DefaultController::class, 'get'])
+                  ->name('generator'),
+             Route::post('/{generator}/preview')
```
Comment (arogachev) [inline config/routes.php:40]: Why `POST` is used? Seems like nothing is modified, so `GET` would be more appropriate?

File: src/Controller/DefaultController.php [hunk 11]
```diff
@@ -4,180 +4,108 @@
...
+        $validationResult = $generator->validate();
+        if ($validationResult->isValid()) {
+            $files = $generator->generate();
+            if ($file !== null) {
+                foreach ($files as $generatedFile) {
+                    if ($generatedFile->getId() === $file) {
+                        $content = $generatedFile->preview();
+                        if ($content !== false) {
+                            return $this->responseFactory->createResponse(['content' => $content]);
```
Comment (arogachev) [inline src/Controller/DefaultController.php:69]: High nesting level (5 levels). Maybe refactor using early return for example?

File: src/Generator/AbstractGenerator.php [hunk 12]
```diff
@@ -58,7 +55,6 @@ abstract class AbstractGenerator implements GeneratorInterface, DataSetInterface
 
     public function __construct(
         protected Aliases $aliases,
-        protected View $view,
         protected ValidatorInterface $validator,
     ) {
     }
```
Comment (arogachev) [inline src/Generator/AbstractGenerator.php:60]: `formView()` method is no longer used and can be deleted (line 143).

**Review by arogachev — CHANGES_REQUESTED** [review]
:+1: 

By the way, what about tests?

**Comment by xepozz** [comment]
Suggest not to do a lot of additional refactoring because I would do the #55 issue instead.

**Review by samdark — APPROVED** [review]

**Pull request merged**